lib/check_graphite/projection.rb

```diff
@@ -0,0 +1,39 @@
+require 'attime'
+require 'bigdecimal'
+require 'nagios_check'
+require 'linear-regression'
+
+module CheckGraphite
+  module Projection
+    def self.included(base)
+      base.on '--projection FUTURE_TIMEFRAME', :default => '2days' do |timeframe|
+        options.send(:processor=, method(:projected_value))
+        options.send(:timeframe=, timeframe)
+      end
+    end
+
+    def projected_value(datapoints)
+      ys, xs = datapoints.transpose
+      lr = Regression::Linear.new(xs, ys)
+      future = CheckGraphite.attime(options.timeframe, xs[-1])
+      value = lr.predict(future.to_i)
+      p = Regression::CorrelationCoefficient.new(xs, ys).pearson
+      store_value options.name, value
+      # Unfortunately, nagios_check converts both primary and
+      # secondary values to float. Hence manual assignment instead:
+      @values['p-value'] = format_float(p.abs)
+      store_message "#{options.name}=#{format_float(value)} in #{options.timeframe}"
+      return value
+    end
+
+    private
+
+     def format_float(v)
+       if v.nan?
+         'undefined'
+       else
+         BigDecimal.new(v, 3).to_s('F')
```

Comment:
This thing needs a test. It needs to be documented why the formatting uses three significant digits (and not five, three or 99). Also, why not for example `sprintf('%.3f', v)` (because I assume this method only ever expects numbers in the 0.0-1.0 range)?

This line has actually been bugging me for a few days, but it was not until today when it just popped into my head that I realised why. It's one of those things that you kind of gloss over, like "yeah, sure and then there's some formatting". But it's not just formatting, it's very deliberate formatting, but with no test to motivate it.